ERCS/erc-7087.md

```diff
@@ -0,0 +1,163 @@
+---
+eip: 7087
+title: New features for Web3 URL in Auto Mode
+description: Add features to Web3 URL in auto mode
+author: Qi Zhou (@qizhou), Nicolas Deschildre (@nand2)
+discussions-to: https://ethereum-magicians.org/t/erc-7087-mime-type-for-web3-url-in-auto-mode/14471
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-05-28
+requires: 6860
+---
+
+## Abstract
+
+This standard extends the [ERC-6860](./eip-6860.md) `web3://` standard in auto mode with new features :
+
+- Determine the MIME type of a URL via additional query parameters.
+- Ability to name fields when ``?returns=`` is used.
+
+## Motivation
+
+The web3 URL standard defines a way to determine the MIME type of a link in auto mode via the suffix of the last argument. However, most of existing contracts on Ethereum do not expect such suffix and thus the protocol cannot determine and return the right MIME type (e.g., contracts returning SVG images with token id as parameter). The standard proposes additional query parameters for a Web3 URL to determine the MIME type.
+
+Additionally, when ``?returns=`` is used with returning complex structured data, an usual step after fetching the URL return is to map the returned arrays into proper named structs. To skip this tedious task, this standard propose a way to name the fields directly in ``?returns=``.
```

Comment:
When you say "determine", do you mean "indicate"? As in the URL indicates to the HTTP server what mime type to return?

Reply:
Yes indeed, "indicate" when the returned data MIME type is implicit (not returned by the smartcontract) -- so in this case an additional query parameter in the URL indicate what is the MIME type to be returned as ``content-type`` when returning the result.

But additonally, in the case of the data returned being a data URL, "determine" by having an additional query parameter indicate that the data is a data URL, so that the MIME type can be extracted from the returned data.

I have rewritten the abstract and motivation sections, hopefully making all this clearer.